src/main/resources/global/90-init-alfresco.sh

```diff
@@ -29,35 +26,22 @@ fi
 if [ -n "$CATALINA_HOME" ]; then
 
   user="tomcat"
-  if [[ ($(stat -c %U /opt/alfresco/alf_data) != "$user" && $(stat -c %a /opt/alfresco/alf_data) -lt 766) ]]; then
-    if [[ "$ENABLE_CHOWNCUTOFF" == 'true' ]]; then
-        # check if number of files exceeds cutoff value to prevent longrunning chown process.
-        find /opt/alfresco/alf_data | head -n $CHOWNCUTOFF > /dev/null
-        # * pipestatus will be 0 if the above find ended before head ended.
-        # * pipestatus will be 141 if the above head ends before find -which implies there are more than CHOWNCUTOFF files,
-        #   which means the following chown would be expensive in time.
-        #   sigpipe will cause the find to end.
-        if [[ ${PIPESTATUS[0]} -eq 0 ]]; then
-          chown -R $user:$user /opt/alfresco/alf_data
-        else
-          # custom exit code for debug to this script
-          exit 64
-        fi
-    else
-        chown -R $user:$user /opt/alfresco/alf_data
-    fi
+  # Check existence of alf_data/contentstore
+  # -> if dir exists, check if the permissions are incorrect
+  # --> if permissions are incorrect, exit with error code
+  # -> else if not exists, chown only alf_data
+  if [[ -e /opt/alfresco/alf_data/contentstore \
+   && ($(stat -c %U /opt/alfresco/alf_data) != "$user" && $(stat -c %a /opt/alfresco/alf_data) -lt 766) ]]; then
```

Comment:
First, we should check for /opt/alfresco/alf_data/contentstore.deleted as well, we need both to be present.
Second, I think this should be `stat -c %U /opt/alfresco/alf_data/contentstore`, and likewise for `stat -c %a` and for `contentstore.deleted`. We don't need permissions on alf_data itself if both contentstore and contentstore.deleted exist and we have permissions on them.